### PR TITLE
feat: rename `@sequelize/sqlite` to `@sequelize/sqlite3`, `@sequelize/ibmi` to `@sequelize/db2-ibmi`, ban conflicting options

### PR DESCRIPTION
**Pull request by ephys**
## Description of Changes

This PR:

- renames `@sequelize/sqlite` to `@sequelize/sqlite3`
- renames `@sequelize/ibmi` to `@sequelize/db2-ibmi`
- bans dialect-specific options that conflict with Sequelize
  - mariadb's `arrayParenthesis`, `autoJsonMap`, `checkNumberRange`, `permitSetMultiParamEntries`
  - mssql's `enableAnsiNullDefault`
  - mysql's `nestTables`, `stringifyObjects`, `queryFormat`

Notes:

SQLite's internal dialect name was changed to `sqlite3` in preparation for `better-sqlite3`, but ibmi's internal dialect name remains `ibmi` even though the package is named `db2-ibmi` as the name is less verbose

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "sscce-mysql": "cross-env DIALECT=mysql yarn sscce",
```
Comment (WikiRik) [inline package.json:59]: I can't comment on line 21, but that should be updated as well

File: packages/core/package.json [hunk 2]
```diff
@@ -98,7 +98,7 @@
   "keywords": [
     "mysql",
     "mariadb",
-    "sqlite",
+    "sqlite3",
```
Comment (WikiRik) [inline packages/core/package.json:101]:
```suggestion
    "sqlite",
    "sqlite3",
```

**Review by WikiRik — COMMENTED** [review]
I checked the amount of tests run for sqlite and they are unchanged

File: package.json (hunk 1 above)
Comment (ephys) [reply package.json:59]: Done

File: packages/core/package.json (hunk 2 above)
Comment (ephys) [reply packages/core/package.json:102]: Done

**Review by WikiRik — APPROVED** [review]

**Pull request merged**